test/tests/pbs_admin_suspend.py

```diff
@@ -405,3 +413,416 @@ def test_degraded_resv(self):
+        try:
+            self.server.expect(
+                NODE, {'maintenance_jobs': ""}, id='vn[0]', max_attempts=10)
+        except Exception, e:
+            self.assertFalse(
+                e.rv, "FAIL:Node vn[0] has maintenance_jobs set")
+            print ("PASS:Node vn[0] do not have maintenance_jobs set")
+
+        # Regular suspend a job
+        self.server.sigjob(jid2, 'suspend', runas=ROOT_USER)
+
+        # List maintenance_jobs again
+        try:
+            self.server.expect(
+                NODE, {'maintenance_jobs': ""}, id='vn[0]', max_attempts=10)
+        except Exception, e:
+            self.assertFalse(
+                e.rv, "FAIL:Node vn[0] has maintenance_jobs set")
+            print ("PASS:Node vn[0] do not have maintenance_jobs set")
+
+    def test_preempt_order(self):
+        """
+        Test that sched_preempt_order has no impact
+        on admin-suspend
+        """
+
+        # set preempt_order to R
+        self.scheduler.set_sched_config({'preempt_order': 'R'})
+
+        # submit a job
+        j = Job(TEST_USER)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid)
+
+        # admin-suspend job and it is in S state
+        self.server.sigjob(jid, 'admin-suspend')
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+
+        # admin-resume the job
+        self.server.sigjob(jid, 'admin-resume')
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid)
+
+    def test_hook(self):
+        """
+        List maintenance_jobs via hook
+        """
+
+        # Create and import a hook
+        hook_name = "test"
+        hook_body = """
+import pbs
+
+vn = pbs.server().vnode('vn[0]')
+pbs.logmsg(pbs.LOG_DEBUGa,\
+"list of maintenance_jobs are %s" % vn.maintenance_jobs)
+"""
+
+        a = {'event': 'exechost_periodic', 'enabled': 'True', 'freq': 5}
+        self.server.create_import_hook(hook_name, a, hook_body)
+
+        # submit few jobs
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state=R': 2})
+
+        # wait for the periodic hook
+        time.sleep(5)
+
+        # look for the log message
+        self.mom.log_match("list of maintenance_jobs are None", max_attempts=5)
+
+        # admin-suspend jobs
+        self.server.sigjob(jid1, 'admin-suspend')
+        self.server.sigjob(jid2, 'admin-suspend')
+
+        # wait for periodic hook and check mom_log
+        time.sleep(5)
+        self.mom.log_match("list of maintenance_jobs are %s" %
+                           (jid1 + "," + jid2), max_attempts=5)
+
+        # admin-resume job1
+        self.server.sigjob(jid1, 'admin-resume')
+
+        # wait for periodic hook and check mom_log
+        time.sleep(5)
+        self.mom.log_match(
+            "list of maintenance_jobs are %s" % (jid2), max_attempts=5)
```

Comment:
Al pointed out to me once that the correct syntax for a tuple with one item is with a trailing comma.  Something like (jid2,).  You might want to talk to Al about this.